### PR TITLE
fix: avoid PRs when there's no contributions

### LINKED ISSUE #31: When a user is found, but the bot cannot determine contributions, tell the user
**Is your feature request related to a problem? Please describe.**
Sometimes when asking the bot to add contributions it won't be able to determine the types of contributions, this does not stop the bot opening a PR with an empty array of contributions see https://github.com/all-contributors/all-contributors-bot/pull/29/files#diff-c47fb5a6d4fde805d2706fc934e8d354R38

**Describe the solution you'd like**
When a user is found, but the bot cannot determine contributions, tell the user. A PR should not be opened.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Berkmann18**
This should resolve #31.
## How?
By checking, right before submitting a PR and commenting, if the `contributions` array is empty or not.
If it is, it will comment accordingly otherwise proceed as usual.

### CODE REVIEW

File: src/processIssueComment.js [hunk 1]
```diff
@@ -42,18 +42,24 @@ async function processAddContributor({
...
 
-    const pullRequestURL = await repository.createPullRequestFromFiles({
-        title: `docs: add ${who} as a contributor`,
-        body: `Adds @${who} as a contributor for ${contributions.join(
-            ', ',
-        )}.\n\nThis was requested by ${commentReply.replyingToWho()} [in this comment](${commentReply.replyingToWhere()})`,
-        filesByPath: filesByPathToUpdate,
-        branchName: `all-contributors/add-${who}`,
-    })
+    if (contributions.length) {
```
Comment (jakebolam) [inline src/processIssueComment.js]: Could we use a guard approach instead? https://link.medium.com/klVQCctMvT

File: src/processIssueComment.js [hunk 2]
```diff
@@ -42,18 +42,24 @@ async function processAddContributor({
...
 
-    commentReply.reply(
-        `I've put up [a pull request](${pullRequestURL}) to add @${who}! :tada:`,
-    )
+        commentReply.reply(
+            `I've put up [a pull request](${pullRequestURL}) to add @${who}! :tada:`,
+        )
+    } else {
+        commentReply.reply(
+            `I could not find any contributions associated to @${who}! :sod:`,
```
Comment (jakebolam) [inline src/processIssueComment.js]: Like the use of emojis.

I think what we're after here is `i couldn't determine any contributions to add, did you specify contributions?`

File: test/ContentFiles/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -18,7 +18,7 @@ Thanks goes to these wonderful people ([emoji key](https://github.com/all-contri
...
 <!-- prettier-ignore -->
-| [<img src=\\"https://avatars2.githubusercontent.com/u/3534236?v=4\\" width=\\"100px;\\"/><br /><sub><b>Jake Bolam</b></sub>](https://jakebolam.com)<br />[💻](#code-jakebolam \\"Code\\") [🤔](#ideas-jakebolam \\"Ideas, Planning, & Feedback\\") [🚇](#infra-jakebolam \\"Infrastructure (Hosting, Build-Tools, etc)\\") [⚠️](#test-jakebolam \\"Tests\\") | [<img src=\\"https://avatars2.githubusercontent.com/u/7265547?v=4\\" width=\\"100px;\\"/><br /><sub><b>tbenning</b></sub>](https://github.com/tbenning)<br />[🎨](#design-tbenning \\"Design\\") |
+| [<img src=\\"https://avatars2.githubusercontent.com/u/3534236?v=4\\" width=\\"100px;\\" alt=\\"Jake Bolam\\"/><br /><sub><b>Jake Bolam</b></sub>](https://jakebolam.com)<br />[💻](#code-jakebolam \\"Code\\") [🤔](#ideas-jakebolam \\"Ideas, Planning, & Feedback\\") [🚇](#infra-jakebolam \\"Infrastructure (Hosting, Build-Tools, etc)\\") [⚠️](#test-jakebolam \\"Tests\\") | [<img src=\\"https://avatars2.githubusercontent.com/u/7265547?v=4\\" width=\\"100px;\\" alt=\\"tbenning\\"/><br /><sub><b>tbenning</b></sub>](https://github.com/tbenning)<br />[🎨](#design-tbenning \\"Design\\") |
```
Comment (jakebolam) [inline test/ContentFiles/__snapshots__/index.test.js.snap]: I'm having difficulty seeing what changed here.

File: src/processIssueComment.js (hunk 1 above)
Comment (Berkmann18) [reply src/processIssueComment.js]: Sure.

File: src/processIssueComment.js (hunk 2 above)
Comment (Berkmann18) [reply src/processIssueComment.js]: I've changed the comment to your suggestion.

File: test/ContentFiles/__snapshots__/index.test.js.snap (hunk 3 above)
Comment (Berkmann18) [reply test/ContentFiles/__snapshots__/index.test.js.snap]: It added `alt='...'`.

File: src/processIssueComment.js [hunk 4]
```diff
@@ -18,6 +18,12 @@ async function processAddContributor({
     who,
     contributions,
 }) {
+    if (!contributions.length) {
+        return commentReply.reply(
+            `I couldn't determine any contributions to add, did you specify any contributions?`,
```
[the remark this answers is not in the document]
Comment (Berkmann18) [reply src/processIssueComment.js]: Sure.
In the meantime, I'll use this URL: https://github.com/all-contributors/all-contributors#emoji-key-

File: src/processIssueComment.js [hunk 5]
```diff
@@ -18,6 +18,12 @@ async function processAddContributor({
     who,
     contributions,
 }) {
+    if (!contributions.length) {
```
[the remark this answers is not in the document]
Comment (Berkmann18) [reply src/processIssueComment.js]: For coverage purposes, yes.

File: test/processIssueComment.test.js [hunk 6]
```diff
@@ -35,6 +36,31 @@ describe('Process Issue Comment', () => {
         )
         expect(mockContext.github).not.toHaveBeenCalled()
     })
+
+    // test('If there is nil contributions, inform about it', async () => {
```
Comment (jakebolam) [inline test/processIssueComment.test.js]: This test looks good to me!

Comment (Berkmann18) [reply test/processIssueComment.test.js]: The only thing is that the part of `parseIssueComment` being tested doesn't work properly if there's no actual issue/comment being linked to it and I'm unsure if whether the function is missing some data we need for the mock testing or if there's a way to simulate a real issue where the comment is being posted in.

Comment (jakebolam) [reply test/processIssueComment.test.js]: I think its fine to add another 'export' from this file to make testing easier (so we can test `processAddContributor` directly.

File: src/processIssueComment.js [hunk 7]
```diff
@@ -24,6 +24,14 @@ async function processAddContributor({
     contributions,
     defaultBranch,
 }) {
+    if (!contributions.length) {
+        context.log.debug('No contributions')
+        return commentReply.reply(
+            `I couldn't determine any contributions to add, did you specify any contributions?
+            Please make sure to use [valid contribution names](https://github.com/all-contributors/all-contributors#emoji-key-).`,
```
Comment (jakebolam) [inline src/processIssueComment.js]: 👍

File: test/processIssueComment.test.js (hunk 6 above)
Comment (jakebolam) [reply test/processIssueComment.test.js]: We have a few of the end-to-end tests in here: https://github.com/all-contributors/all-contributors-bot/blob/master/test/tasks/processIssueComment/probot-app-e2e.test.js. Generally we try to reserve end-to-end tests for the core use cases. You could add one here too tho.

**Comment by Berkmann18** [comment]
@jakebolam Not for now, I've noticed the relevant module was moved and stuff so currently to fix post-`rebase`  test failures (while dealing with personal stuff).

**Comment by jakebolam** [comment]
no worries @Berkmann18. Let me know if you need anything. 

FYI: GitHub shows conflicts, but hopefully using git locally shouldn't be too much of an issue. Was working on an out of date branch earlier, and git auto-merged it 🎉

**Comment by Berkmann18** [comment]
@jakebolam Did you encountered any `Jest encountered an unexpected token` errors when running `yarn test`?
I've been trying to rebase the branch with `master` and despite having either:
```js
module.exports = { processIssueCommentApp }
//Or
module.exports = processIssueCommentApp //like on master
```
It seems to be failed due to something I'm oblivious about.

File: test/processIssueComment.test.js (hunk 6 above)
Comment (Berkmann18) [reply test/processIssueComment.test.js]: I've had a look at that test file and the issue thread it uses but I don't know how I could go about making sure it looks at the right comment (as it seems to look at the whole thread).

**Comment by Berkmann18** [comment]
Okay, I'll open a new PR with the changes I did from here (without the additional tests which I left on the working copy as I'm unsure how `nock` gets the correct comment from the whole thread in #1.

**Pull request closed without merging**